assignment/core/src/main/java/com/assignment/core/services/Impl/NodeDeletionServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.assignment.core.services.Impl;
+
+import com.assignment.core.services.NodeDeletionService;
+import org.apache.sling.api.resource.*;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.RepositoryException;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Implementation of the NodeDeletionService interface that deletes nodes under the specified root path.
+ */
+@Component(service = NodeDeletionService.class)
+public class NodeDeletionServiceImpl implements Runnable, NodeDeletionService {
+
+    private static final Logger log = LoggerFactory.getLogger(NodeDeletionServiceImpl.class);
+
+    @Reference
+    private ResourceResolverFactory resourceResolverFactory;
+
+    private String rootPath;
+
+    @Override
+    public void run() {
```

Comment:
Use this run() method in you scheduler class not in the service, it will not run.